### PR TITLE
Full json rendering

### PR DESCRIPTION
**Pull request by klokane**
There are changes required for Full rendering

Side effects:
- is complete remove "renderFlag" from Element
- remove cross dependency between Compact and Full Serializer

Actually checking all related changes in tests. (Good message - It seems to be OK)

There is "do not merge" flag until merge updated test result

### CODE REVIEW

File: src/Serialize.cc [hunk 1]
```diff
@@ -19,9 +19,9 @@ namespace drafter {
     {
         refract::ArrayElement* array = new refract::ArrayElement;
 
-        if (rFull) {
-            value->renderType(refract::IElement::rFull);
-        }
+        //if (rFull) {
+        //    value->renderType(refract::IElement::rFull);
+        //}
```
Comment (klokane) [inline src/Serialize.cc]: remove commented code

File: src/refract/ExpandVisitor.cc [hunk 2]
```diff
@@ -214,7 +214,7 @@ namespace refract
             if (IElement* referenced = registry.find(href->value)) {
                 referenced = ExpandOrClone(referenced);
                 MetaIdToRef(*referenced);
-                referenced->renderType(IElement::rFull);
+                //referenced->renderType(IElement::rFull);
```
Comment (klokane) [inline src/refract/ExpandVisitor.cc]: remove commented code

File: src/refract/JSONSchemaVisitor.cc [hunk 3]
```diff
@@ -33,7 +33,7 @@ namespace refract
             RenderJSONVisitor v;
             Visit(v, *(*it));
             IElement *e = v.getOwnership();
-            e->renderType(IElement::rCompact);
+            //e->renderType(IElement::rCompact);
```
Comment (klokane) [inline src/refract/JSONSchemaVisitor.cc]: remove commented code

File: src/refract/JSONSchemaVisitor.cc [hunk 4]
```diff
@@ -153,10 +150,7 @@ namespace refract
 
     void JSONSchemaVisitor::addMember(const std::string& key, IElement *val)
     {
-        pObj->push_back(new MemberElement(
-                            key,
-                            val,
-                            IElement::rCompact));
+        pObj->push_back(new MemberElement( key, val));
```
Comment (pksunkara) [inline src/refract/JSONSchemaVisitor.cc]: `( key`. Bad extra space.

**Comment by pksunkara** [comment]
Do we still need the SerializeCompactVisitor?

**Comment by w-vi** [comment]
Yes, because JSON and JSONSchema Visitors are serializing into compact. It should be replaced one day with ApplyVisitor.

**Comment by klokane** [comment]
In general, both serializers should be replaced by ApplyVisitor  :)

**Comment by klokane** [comment]
I go through tests results and everything seems to be correct
Someone hero to merge into master? :)

**Pull request merged**